dockerfiles/grist/ressources/dinum-custom.js

```diff
@@ -0,0 +1,143 @@
+    const dateFormatter = new Intl.DateTimeFormat('fr-FR', {
+      dateStyle: 'full',
+      timeStyle: 'short',
+      timeZone: 'Europe/Paris',
+    });
+    const timeFormatter = new Intl.DateTimeFormat('fr-FR', {
+      timeStyle: 'short',
+      timeZone: 'Europe/Paris',
+    });
+
+    dialog.innerHTML = `
+        <p>
+        Une maintenance de Grist est prévue le ${dateFormatter.format(maintainanceStartDate)} heure de Paris
+          jusqu'à ${timeFormatter.format(maintainanceEndDate)}, période durant laquelle le service sera momentanément indisponible. <br>
+        Pour toute remarque ou question, merci de nous contacter via <a href="mailto:donnees@anct.gouv.fr">donnees@anct.gouv.fr</a></p>
+        <p>Merci de votre compréhension</p>
+        <form method="dialog">
+          <p>
+            <input type="checkbox" id="jaiCompris">
+            <label for="jaiCompris">J'ai lu et compris, merci de ne plus me montrer ce message</label>
+          </p>
+          <button disabled id="fermerPopinMaintenance">Fermer</button>
+        </form>
+    `;
+    document.body.appendChild(dialog);
+    document.getElementById('fermerPopinMaintenance').onclick = function onMessageUnderstood() {
+      localStorage.maintainanceStartDateAgreement = maintainanceStartDate.toISOString();
+    }
+    document.getElementById('jaiCompris').onchange = function (ev) {
+      document.getElementById('fermerPopinMaintenance').disabled = !ev.target.checked;
+    }
+    dialog.showModal();
+  }
+
+  if (Date.now() < maintainanceEndDate.getTime() &&
+    (!localStorage.maintainanceStartDateAgreement || localStorage.maintainanceStartDateAgreement !== maintainanceStartDate.toISOString())) {
+    window.addEventListener('load', showDialog);
+  }
+})();
+
+// ========================
+// END dialog
+// ========================
+
+// ========================
+// START gauffre
+// ========================
+
+window.onload = (event) => {
+  const pathname = window.location.pathname;
+
+  // Gauffre is only inserted on grist home page
+  if (pathname !== '/o/docs/'){
+    return 1;
+  }
+
+
+  waitForElm('.test-top-header').then((elm) => {
+    // Create gauffre button Tag
+    const gristBar = document.getElementsByClassName('test-top-header')[0];
+    const gauffreDiv = document.createElement('div');
+    const gauffreButton = document.createElement('button');
+    const gauffreText = "Les services de La Suite numérique";
+
+    gauffreDiv.className = 'gauffre-container';
+    gauffreButton.type = "button";
+    gauffreButton.className = "lasuite-gaufre-btn lasuite-gaufre-btn--vanilla js-lasuite-gaufre-btn";
+    gauffreButton.title = gauffreText;
+    gauffreButton.text = gauffreText;
+    gauffreDiv.appendChild(gauffreButton);
+    gristBar.insertBefore(gauffreDiv, gristBar.lastChild);
+
+    // Create gauffre script Tag
+    const gauffreScript = document.createElement('script');
+    const head = document.getElementsByTagName('head')[0];
```

Comment:
Nitpicking: You may remove that variable and rather use `document.head` instead.

Reply:
fixed in https://github.com/numerique-gouv/dockerfiles/pull/5/commits/45ec7acd23c208efa1bc45e47b1eb53e9b40eacf